src/wp-includes/media.php

```diff
@@ -5661,6 +5661,27 @@ function wp_get_loading_optimization_attributes( $tag_name, $attr, $context ) {
 			}
 	}
 
+	/**
+	 * Filters whether to short-circuit loading optimization attributes.
+	 *
+	 * Returning an array from the filter will effectively short-circuit the loading of optimization attributes,
+	 * returning that value instead.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param false|array $loading_attrs The loading optimization attributes.
+	 * @param array       $attr          Array of the attributes for the tag.
+	 * @param string      $context       Context for the element for which the loading optimization attribute is requested.
+	 */
+	$loading_attrs = apply_filters( 'wp_pre_get_loading_optimization_attributes', false, $attr, $context );
```

Comment:
Two points of feedback here:
* The `$tag_name` value is missing and needs to be passed to the filter as it is critical to make loading optimization changes.
* It's more common in core to use the full function name as filter name and prefix that with "pre_", i.e. `pre_wp_` instead of `wp_pre_`.
```suggestion
	 * @param false|array $loading_attrs The loading optimization attributes.
	 * @param string      $tag_name      The tag name.
	 * @param array       $attr          Array of the attributes for the tag.
	 * @param string      $context       Context for the element for which the loading optimization attribute is requested.
	 */
	$loading_attrs = apply_filters( 'pre_wp_get_loading_optimization_attributes', false, $tag_name, $attr, $context );
```

Reply:
Addition to Felix changes please add default value for the `$loading_attrs`.

```
* @param false|array $loading_attrs The loading optimization attributes. Default false.
```

Reply:
@mukeshpanchal27 I don't think that's correct. For a pre-filter, we need to use a clearly different type. If someone wants to override this to return an empty array, they should be able to do so, which with the new change is no longer possible. So I think this should be changed back to how it was, using `false|array` and `false` as the default.